--- webrtc-audio-processing-1.0/meson.build~	2021-12-01 22:21:02.000000000 +0100
+++ webrtc-audio-processing-1.0/meson.build	2021-12-01 23:46:59.069767222 +0100
@@ -4,7 +4,7 @@
   default_options : [ 'warning_level=1',
                       'buildtype=debugoptimized',
                       'c_std=c11',
-                      'cpp_std=c++14',
+                      'cpp_std=c++17',
                     ]
 )
 
From 6064932abf662db451defe7326fc2dd0c3016f99 Mon Sep 17 00:00:00 2001
From: Arun Raghavan <arun@asymptotic.io>
Date: Tue, 19 Oct 2021 16:01:05 -0400
Subject: [PATCH] Add missing header for C++17 compatibility

Hopefully we can drop this change with the next update.
---
 webrtc/modules/audio_processing/aec3/reverb_frequency_response.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/webrtc/modules/audio_processing/aec3/reverb_frequency_response.h b/webrtc/modules/audio_processing/aec3/reverb_frequency_response.h
index b164186..84ea441 100644
--- a/webrtc/modules/audio_processing/aec3/reverb_frequency_response.h
+++ b/webrtc/modules/audio_processing/aec3/reverb_frequency_response.h
@@ -12,6 +12,7 @@
 #define MODULES_AUDIO_PROCESSING_AEC3_REVERB_FREQUENCY_RESPONSE_H_
 
 #include <array>
+#include <memory>
 #include <vector>
 
 #include "absl/types/optional.h"
-- 
GitLab

